huaweicloud/services/as/data_source_huaweicloud_as_configurations.go

```diff
@@ -0,0 +1,231 @@
+															"ip_type": {
+																Type:     schema.TypeString,
+																Computed: true,
+															},
+															"bandwidth": {
+																Type:     schema.TypeList,
+																Computed: true,
+																Elem: &schema.Resource{
+																	Schema: map[string]*schema.Schema{
+																		"size": {
+																			Type:     schema.TypeInt,
+																			Computed: true,
+																		},
+																		"share_type": {
+																			Type:     schema.TypeString,
+																			Computed: true,
+																		},
+																		"charging_mode": {
+																			Type:     schema.TypeString,
+																			Computed: true,
+																		},
+																	},
+																},
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+									"user_data": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"metadata": {
+										Type:     schema.TypeMap,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceASConfigurationRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	asClient, err := conf.AutoscalingV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating autoscaling client: %s", err)
+	}
+
+	opts := configurations.ListOpts{
+		Name:    d.Get("scaling_configuration_name").(string),
+		ImageID: d.Get("image_id").(string),
+	}
+	page, err := configurations.List(asClient, opts).AllPages()
+	if err != nil {
+		return diag.Errorf("error getting AS Configuration list: %s", err)
+	}
+
+	configurationList, err := page.(configurations.ConfigurationPage).Extract()
+	if err != nil {
+		return diag.Errorf("error extract to AS Configuration list: %s", err)
+	}
+
+	if len(configurationList) == 0 {
+		return fmtp.DiagErrorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+
+	ids := make([]string, 0, len(configurationList))
+	elements := make([]map[string]interface{}, 0, len(configurationList))
+	for _, configuration := range configurationList {
+		configurationMap := map[string]interface{}{
+			"scaling_configuration_name": configuration.Name,
+			"instance_config":            flattenInstanceConfig(configuration.InstanceConfig),
+			"status":                     normalizeConfigurationStatus(configuration.ScalingGroupID),
+		}
+		ids = append(ids, configuration.ID)
+		elements = append(elements, configurationMap)
+	}
+
+	d.SetId(hashcode.Strings(ids))
+	mErr := multierror.Append(nil,
+		d.Set("configurations", elements),
+		d.Set("region", conf.GetRegion(d)),
```

Comment:
the param of region has been otained, it is better to use it directly here

Reply:
OK